Composer/packages/client/src/CreationFlow/DefineConversation/index.js

```diff
@@ -59,6 +75,22 @@ export function DefineConversation(props) {
     });
   };
 
+  function getDefaultName() {
+    let i = 0;
+    const bot = templateId || botName;
```

Comment:
We are constructing the bot name in two path.

You should use a do-while loop, which should eliminate constructing twice.